### PR TITLE
Make account lookup faster with sync.Map

### PR DESCRIPTION
**Pull request by derekcollison**
This switches account lookup to use sync.Map. This should help in concurrent access from routes and gateways when interest graph churn forces cache misses.

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -98,8 +98,8 @@ type Server struct {
 	listener         net.Listener
 	gacc             *Account
 	sys              *internal
-	accounts         map[string]*Account
-	activeAccounts   int
+	accounts         sync.Map
+	activeAccounts   int32
```
Comment (kozlovic) [inline server/server.go]: Looks like issue https://github.com/golang/go/issues/599 is still opened, so I usually put the fields I want to use atomic ops on at top of structure, and often use 64bits versions. Not sure if this is an issue or not (you would have to try with the linux 386 exe to see if it works or not.

File: server/reload.go [hunk 2]
```diff
@@ -787,17 +787,30 @@ func (s *Server) applyOptions(ctx *reloadContext, opts []option) {
 func (s *Server) reloadAuthorization() {
 	s.mu.Lock()
 
-	oldAccounts := s.accounts
-	s.accounts, s.gacc = nil, nil
+	// We need to drain the old accounts here since we have something
+	// new configured. We do not want s.accounts to change since that would
+	// mean adding a lock to lookupAccount which is what we are tryin to optimize
```
Comment (kozlovic) [inline server/reload.go]: trying

File: server/server.go (hunk 1 above)
Comment (derekcollison) [reply server/server.go]: Meaning it will fail or just be slow?

Comment (derekcollison) [reply server/server.go]: No perf diff on OSX. Will check linux.

File: server/reload.go (hunk 2 above)
Comment (derekcollison) [reply server/reload.go]: will fix

File: server/server.go (hunk 1 above)
Comment (kozlovic) [reply server/server.go]: I think very early reports were that it would core dump on 386.

Comment (derekcollison) [reply server/server.go]: None in linux either.

Comment (derekcollison) [reply server/server.go]: Ah, not sure how to test that atm..

**Comment by kozlovic** [comment]
LGTM! verified that linux 386 was not crashing.

File: server/server.go (hunk 1 above)
Comment (derekcollison) [reply server/server.go]: Thx!

**Pull request merged**